src/sources/host_metrics/mod.rs

```diff
@@ -299,6 +299,7 @@ fn add_collector(collector: &str, mut metrics: Vec<Metric>) -> Vec<Metric> {
     metrics
 }
 
+#[allow(clippy::missing_const_for_fn)]
```

Comment:
This seems unrelated. Should we make something here `const` on master? Feels a little off to add an `allow`.

Reply:
Yeah, agreed. If we need to allow it, could we add a comment about why?

Reply:
if I remove the `allow` I get
```
error: this could be a `const fn`
   --> src/sources/host_metrics/mod.rs:303:1
    |
303 | / fn init_roots() {
304 | |     #[cfg(target_os = "linux")]
305 | |     {
306 | |         use std::sync::Once;
...   |
333 | |     };
334 | | }
    | |_^
    |
note: the lint level is defined here
   --> src/lib.rs:14:9
    |
14  | #![deny(clippy::missing_const_for_fn)] // [nursery] valuable to the optimizer,
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_const_for_fn

error: aborting due to previous error

error: could not compile `vector`
```

Reply:
I'll try to add `const` to the function but I think I tried and it failed to pass the tests on the CI...

Reply:
Ok, seems to work 🤷

Reply:
Fails on linux

```
   Compiling vrl-stdlib v0.1.0 (/home/ubuntu/github-actions-runner/_work/vector/vector/lib/vrl/stdlib)
error[E0013]: constant functions cannot refer to statics
Error:    --> src/sources/host_metrics/mod.rs:309:9
    |
309 |         INIT.call_once(|| {
    |         ^^^^
    |
    = help: consider extracting the value of the `static` to a `const`, and referring to that

error[E0015]: calls in constant functions are limited to constant functions, tuple structs and tuple variants
Error:    --> src/sources/host_metrics/mod.rs:309:9
    |
309 | /         INIT.call_once(|| {
310 | |             match std::env::var_os("PROCFS_ROOT") {
311 | |                 Some(procfs_root) => {
312 | |                     info!(
...   |
330 | |             }
331 | |         });
    | |__________^

error: aborting due to 2 previous errors
Error: aborting due to 2 previous errors
Some errors have detailed explanations: E0013, E0015.
For more information about an error, try `rustc --explain E0013`.
error: could not compile `vector`
Error: could not compile `vector`
To learn more, run the command again with --verbose.
make: *** [Makefile:557: test-shutdown] Error 101
Error: Process completed with exit code 2.
```